types/phoenix_live_view/socket_options.d.ts

```diff
@@ -0,0 +1,206 @@
+import { HooksOptions } from "./hooks";
+import { UploadersOptions } from "./upload_entry";
+import View from "./view";
+
+export type ConnectFunction = (el: HTMLElement) => object;
+export type ConnectParams = object | ConnectFunction;
+
+export interface Defaults {
+  debounce: number;
+  throttle: number;
+}
```

Comment:
Three cheers for using these as a `Partial` so we can get rid of the visual noise from making every field optional. 👍

Reply:
Right-o. I saw that `@types/phoenix` is using `Partial<T>`. I'm still learning TS, but this seems to be better than adding question marks everywhere...